node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -213,6 +221,18 @@ impl<C: Context> ActiveValidator<C> {
         vec![Effect::NewVertex(ValidVertex(Vertex::Vote(proposal_vote)))]
     }
 
+    /// Registers the finalization of a block and checks whether the round exponent should be
+    /// changed
+    pub(crate) fn handle_finalized_block(
+        &mut self,
+        block_timestamp: Timestamp,
+        finalization_timestamp: Timestamp,
+    ) {
+        self.next_round_exp = self
```

Comment:
Either in `round_exp` or here we need to make sure we don't go below `min_round_exp`.